### PR TITLE
Transformation particle to provide set slots provided by hosted particle

### PR DESCRIPTION
**Pull request by mmandlis**
Fixes: https://github.com/PolymerLabs/arcs/issues/1009

### CODE REVIEW

File: runtime/dom-context.js [hunk 1]
```diff
@@ -101,7 +105,7 @@ class DomContext {
     // TODO(sjmiles): remember that attribute names from HTML are lower-case
     return node[name] || node.getAttribute(name);
   }
-  initInnerContexts(slotSpec) {
+  initInnerContexts(slotSpec, globalSubId) {
```
Comment (sjmiles) [inline runtime/dom-context.js]: IMO this method is too complex, suggest factoring it.

File: runtime/dom-slot.js [hunk 2]
```diff
@@ -73,9 +73,12 @@ class DomSlot extends Slot {
...
     return observer;
   }
-
   __initMutationObserver() {
-    const observer = new MutationObserver(async () => {
+    const observer = new MutationObserver(async (records) => {
+      if (!records.some(r => this.getContext().isDirectInnerSlot(r.target))) {
+        return;
+      }
+
```
Comment (sjmiles) [inline runtime/dom-slot.js]: Blank lines typically indicate a new 'step' in the method, I take these as indicator of places where factoring can occur, so a particular method is always a contiguous list of steps (i.e. any step that is complicated to want a line-break is a candidate for factoring into a method). There is a theme here: embracing extreme factoring was a major step in my growth as a developer.

File: runtime/dom-context.js (hunk 1 above)
Comment (mmandlis) [reply runtime/dom-context.js]: done

File: runtime/dom-slot.js (hunk 2 above)
Comment (mmandlis) [reply runtime/dom-slot.js]: i did a small improvement here for now. i'm planning a bigger change re how observers are created and handled, but i prefer to make it in a separate pr, if that's ok with you.

Comment (sjmiles) [reply runtime/dom-slot.js]: Sure thing, my notes were just advisory. Approved!

**Comment by shaper** [comment]
Something about this change seems to break Social post creation -- the EditPost particle fails to render its content into the DetailSlider. I will file an issue with the error detail.

**Pull request merged**